sources/world/README.md

```diff
@@ -1,45 +1,115 @@
 -sort this.properties.region_iso_code \
--o format=topojson prettify admin_regions_lvl2_v2.topo.json
+-o format=topojson prettify ../../data/admin_regions_lvl2_v2.topo.json
 ```
 
 * Convert to GeoJSON (EMS v1)
 ```
-mapshaper -i combine-files 10m_cultural/ne_10m_admin_1_states_provinces.shp \
+mapshaper -i combine-files naturalearth/10m_cultural/ne_10m_admin_1_states_provinces.shp \
 china_states.shp \
 -merge-layers force \
 -filter "this.properties.iso_3166_2 !== ''" \
--filter-fields "name,iso_3166_2" \
--rename-fields "region_iso_code=iso_3166_2,region_name=name" \
+-filter "this.properties.iso_a2 !== '-1'" \
+-filter-fields "name,iso_3166_2,iso_a2,adm0_a3,admin" \
+-rename-fields "region_iso_code=iso_3166_2,region_name=name,country_iso2_code=iso_a2,country_iso3_code=adm0_a3,country_name=admin" \
 -simplify visvalingam interval=1000 keep-shapes \
 -sort this.properties.region_iso_code \
--o rfc7946 format=geojson prettify admin_regions_lvl2_v1.geo.json
+-o rfc7946 format=geojson prettify ../../data/admin_regions_lvl2_v1.geo.json
+```
+
+* Fix for invalid geometries in the GeoJSON file
+```
+node scripts/clean-geom.js -v data/admin_regions_lvl2_v1.geo.json
+```
+
+
+## World Countries
+
+### Getting population and area data from wikidata
+
+Run this query at <https://query.wikidata.org/> and save the result as a CSV.
+
+```spaql
+SELECT 
+?iso2
+(SAMPLE(?pop_value) AS ?pop)
+WHERE {
+ VALUES ?iso2 {
+   "AD" "AE" "AF" "AG" "AI" "AL" "AM" "AO" "AQ" "AR" "AS" "AT" "AU" "AW" "AZ" "BA" "BB" "BD" "BE" "BF" "BG" "BH" "BI" "BJ" "BL" "BM" "BN" "BO" 
+   "NL" "BR" "BS" "BT" "BW" "BY" "BZ" "CA" "CD" "CF" "CG" "CH" "CI" "CK" "CL" "CM" "CN" "TW" "CO" "CR" "CU" "CV" "CW" "CY" "CZ" "DE" "DJ" "DK" 
+   "DM" "DO" "DZ" "EC" "EE" "EG" "EH" "ER" "ES" "ET" "FI" "FJ" "FK" "FM" "FO" "FR" "GA" "GB" "GD" "GE" "GG" "GH" "GI" "GL" "GM" "GN" "GQ" "GR" 
+   "GS" "GT" "GU" "GW" "GY" "HK" "HM" "HN" "HR" "HT" "HU" "ID" "IE" "IL" "IM" "IN" "IO" "IQ" "IR" "IS" "IT" "JE" "JM" "JO" "JP" "KE" "KG" "KH"
+   "KI" "KM" "KN" "KP" "KR" "KW" "KY" "KZ" "LA" "LB" "LC" "LI" "LK" "LR" "LS" "LT" "LU" "LV" "LY" "MA" "MC" "MD" "ME" "MF" "MG" "MH" "MK" "ML" 
+   "MM" "MN" "MO" "MP" "MR" "MS" "MT" "MU" "MV" "MW" "MX" "MY" "MZ" "NA" "NC" "NE" "NF" "NG" "NI" "SX" "NO" "NP" "NR" "NU" "NZ" "OM" "PA" "PE" 
+   "PF" "PG" "PH" "PK" "PL" "PM" "PN" "PS" "PT" "PW" "PY" "QA" "RO" "RS" "RU" "RW" "SA" "SB" "SC" "SD" "SE" "SG" "SH" "SI" "SK" "SL" "SM" "SN" 
+   "SO" "SR" "SS" "ST" "SV" "SY" "SZ" "TC" "TD" "TF" "TG" "TH" "TJ" "TK" "TL" "TM" "TN" "TO" "TR" "TT" "TV" "TZ" "UA" "UG" "UM" "US" "PR" "UY" 
+   "UZ" "VA" "VC" "VE" "VG" "VI" "VN" "VU" "WF" "WS" "XK" "YE" "ZA" "ZM" "ZW"
+ }
+ ?id wdt:P297 ?iso2;
+     wdt:P1082 ?pop_value.
+}
+GROUP BY ?iso2
+ORDER BY ?iso2
+```
+
+### Mapshaper
+
+Dissolve the regions dataset by `country_iso2_code` and join with the Wikidata CSV to generate both the GeoJSON and TopoJSON outputs, each with different simplification thresholds to generate around 2MB datasets.
```

Comment:
Good idea to dissolve the admin regions. The Natural Earth [Admin0 Countries layer](https://www.naturalearthdata.com/downloads/10m-cultural-vectors/10m-admin-0-countries/) includes some unconventional boundaries such as the Baikonur Cosmodrome.

Reply:
Yeah, I think it makes sense to derive the World Countries from the Regions dataset since we have better control of the generation of this dataset.